### ISSUE TITLE
Failure to load all system icons on 2023.03 on windows 11 21H2 (OS Build 22000.1696)

### ISSUE THREAD

**Issue by marlon-sousa** [josephsl/systrayList#3]
Hello,

NVDA key + f11 fails to load the list of systray icons on windows 11. Instead, noothing is returned.

Windows version is 21H2 (OS Build 22000.1696)

Task bar list is being returned normally as far as I know.

Relevant piece of logs upon nvda + f11 trigger follows

```
ERROR - scriptHandler.executeScript (12:25:39.913) - MainThread (7808):
error executing script: <bound method GlobalPlugin.script_createList of <globalPlugins.systrayList.GlobalPlugin object at 0x0721D5F0>> with gesture 'NVDA+f11'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 289, in executeScript
  File "C:\Users\Marlon\AppData\Roaming\nvda\addons\systrayList\globalPlugins\systrayList\__init__.py", line 103, in script_createList
    self._createSystrayList()
  File "C:\Users\Marlon\AppData\Roaming\nvda\addons\systrayList\globalPlugins\systrayList\__init__.py", line 119, in _createSystrayList
    objects = self._findAccessibleLeafsFromWindowClassPath11(path11)
  File "C:\Users\Marlon\AppData\Roaming\nvda\addons\systrayList\globalPlugins\systrayList\__init__.py", line 71, in _findAccessibleLeafsFromWindowClassPath11
    obj = NVDAObjects.IAccessible.getNVDAObjectFromEvent(h, -4, 0).firstChild.children
AttributeError: 'NoneType' object has no attribute 'firstChild'
ERROR - scriptHandler.executeScript (12:26:39.309) - MainThread (7808):
error executing script: <bound method GlobalPlugin.script_createList of <globalPlugins.systrayList.GlobalPlugin object at 0x0721D5F0>> with gesture 'NVDA+f11'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 289, in executeScript
  File "C:\Users\Marlon\AppData\Roaming\nvda\addons\systrayList\globalPlugins\systrayList\__init__.py", line 103, in script_createList
    self._createSystrayList()
  File "C:\Users\Marlon\AppData\Roaming\nvda\addons\systrayList\globalPlugins\systrayList\__init__.py", line 119, in _createSystrayList
    objects = self._findAccessibleLeafsFromWindowClassPath11(path11)
  File "C:\Users\Marlon\AppData\Roaming\nvda\addons\systrayList\globalPlugins\systrayList\__init__.py", line 71, in _findAccessibleLeafsFromWindowClassPath11
    obj = NVDAObjects.IAccessible.getNVDAObjectFromEvent(h, -4, 0).firstChild.children
AttributeError: 'NoneType' object has no attribute 'firstChild'

```

On Windows 10 21H2 (OS Build 19044.2728) the script runs normally.

It seems as if Windows 11 is no longer returning the window from the event you are lokking into.

I cannot say what was the last working version, because I made an update but cannot track easily the previous version.

Still, this seems like a regression.

**Comment by ruifontes** [comment]
Marlon, can you send me the Developer info for navigator object for one of the systray icons?

**Comment by marlon-sousa** [comment]
Hello,

What procedure should I apply to send you this info?

I placed the navigator on an object from systray and pressed nvda + f1.

On Windows 11, here is what it shows

```

INFO - globalCommands.script_navigatorObject_devInfo (12:19:05.859) - MainThread (12496):
Developer info for navigator object:
name: 'NVDA'
role: Role.BUTTON
processID: 11056
roleText: None
states: 
isFocusable: False
hasFocus: False
Python object: <NVDAObjects.UIA.UIA object at 0x0A03EB10>
Python class mro: (<class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: ''
location: RectLTWH(left=3063, top=2256, width=96, height=144)
value: None
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
appModule: <'explorer' (appName 'explorer', process ID 11056) at address 63f7dd0>
appModule.productName: 'Sistema operacional Microsoft® Windows®'
appModule.productVersion: '10.0.22000.1690'
appModule.helperLocalBindingHandle: c_long(18071936)
windowHandle: 131186
windowClassName: 'ToolbarWindow32'
windowControlID: 1504
windowStyle: 1442875981
extendedWindowStyle: 128
windowThreadID: 18012
windowText: 'User Promoted Notification Area'
displayText: ''
UIAElement: <POINTER(IUIAutomationElement) ptr=0x5cd1918 at 95b20d0>
UIA automationID: 
UIA frameworkID: 
UIA runtimeID: (42, 131186, 4, -2147483647, 131186, -4, 7)
UIA providerDescription: [pid:11056,providerId:0x0 Annotation:Microsoft: Annotation Proxy (unmanaged:uiautomationcore.dll); Main(parent link):Microsoft: MSAA Proxy (unmanaged:uiautomationcore.dll)]
UIA className: 
UIA patterns available: LegacyIAccessiblePattern, InvokePattern

```
On windows 10, here is what it shows for this particular button:

```
INFO - globalCommands.script_navigatorObject_devInfo (12:14:17.605) - MainThread (29968):
Developer info for navigator object:
name: 'Dropbox 169.4.5684\nUp to date'
role: Role.BUTTON
processID: 21996
roleText: None
states: 
isFocusable: False
hasFocus: False
Python object: <NVDAObjects.Dynamic_NotificationAreaButtonIAccessible object at 0x080439F0>
Python class mro: (<class 'NVDAObjects.Dynamic_NotificationAreaButtonIAccessible'>, <class 'nvdaBuiltin.appModules.explorer.NotificationArea'>, <class 'NVDAObjects.IAccessible.Button'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=1417, top=1030, width=30, height=50)
value: None
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
appModule: <'explorer' (appName 'explorer', process ID 21996) at address 61a2bb0>
appModule.productName: 'Microsoft® Windows® Operating System'
appModule.productVersion: '10.0.19041.2311'
appModule.helperLocalBindingHandle: c_long(97021744)
windowHandle: 196656
windowClassName: 'ToolbarWindow32'
windowControlID: 1504
windowStyle: 1442876237
extendedWindowStyle: 128
windowThreadID: 24036
windowText: 'User Promoted Notification Area'
displayText: ''
IAccessibleObject: <POINTER(IAccessible) ptr=0x84ff5f8 at 32f0cb0>
IAccessibleChildID: 3
IAccessible event parameters: windowHandle=196656, objectID=-4, childID=3
IAccessible accName: 'Dropbox 169.4.5684\nUp to date'
IAccessible accRole: ROLE_SYSTEM_PUSHBUTTON
IAccessible accState:  (0)
IAccessible accDescription: None
IAccessible accValue: None
```

**Comment by marlon-sousa** [comment]
In windows 10, if I place the navigator object on a systray button and make:

```
nav.parent.name
'User Promoted Notification Area'

nav.parent.parent.name

'User Promoted Notification Area'
nav.parent.parent.parent.name
Nothing returns
```

The same on windows 11 holds interestingly a different result

```
nav.parent.name
'User Promoted Notification Area'

nav.parent.parent.name
''

nav.parent.parent.parent.name
''

nav.parent.parent.parent.parent.name
'Taskbar'

nav.parent.parent.parent.parent.parent.name
'Área de trabalho'
```

If I force yet an upper level, I get a traceback because None object has not name attreibute.

It seems for me though that window hierarchy on Windows 11 has changed a lot from what Windows 10 has.

I, however, am not sure whether this is caused by Windows 11 or by some taskbar organiuzation I might have different om these two machines.

I think this is unlikely. It seems as if in Windows 11 all windows are children of the desktop windows and on Windows 10 this relation is more of a cibling.

However, interestingly running apps are still being retrieved nornally.

**Comment by marlon-sousa** [comment]
I report that windows 11 Version 22H2 (OS Build 22621.1413) is working perfectly.

**Comment by ruifontes** [comment]
Closed.
Works for me and it seems your first Windows 11 was with some problem...
